src/harness/reference_models/pre_iap_filtering/fss_purge.py

```diff
@@ -0,0 +1,301 @@
+  """
+
+  cbsd_antenna_gain = antenna.GetStandardAntennaGains( incidence_angle.hor_cbsd,\
+                       cbsd['registrationRequest']['installationParam']['antennaAzimuth'],
+                       cbsd['registrationRequest']['installationParam']['antennaBeamwidth'],
+                       cbsd['registrationRequest']['installationParam']['antennaGain'])
+
+  return cbsd_antenna_gain
+
+def getMeanPathLossFromCbsdToFss(cbsd, fss):
+  """Get mean ITM path loss from CBSD to FSS protection point.
+
+  Args:
+    cbsd: A CbsdData object
+    fss_entity: The FssProtectionPoint tuple of the FSS.
+  Returns:
+    Mean path loss from CBSD to FSS in dBi.
+  """
+
+  # Extract the CBSD parameters
+  cbsd_latitude = cbsd['registrationRequest']['installationParam']['latitude']
+  cbsd_longitude = cbsd['registrationRequest']['installationParam']['longitude']
+  cbsd_height = cbsd['registrationRequest']['installationParam']['height']
+  cbsd_height_type = cbsd['registrationRequest']['installationParam']['heightType']
+  cbsd_indoor_dep = cbsd['registrationRequest']['installationParam']['indoorDeployment']
+  if cbsd_height_type == 'AMSL':
+      cbsd_height_agl = convertAmslToAgl(cbsd_latitude, cbsd_longitude, cbsd_height)
+  else:
+      cbsd_height_agl = cbsd_height
+
+  # Invoking ITM propagation model
+  db_loss, incidence_angle, _ = wf_itm.CalcItmPropagationLoss(
+                                         cbsd_latitude,
+                                         cbsd_longitude,
+                                         cbsd_height_agl,
+                                         fss.latitude,
+                                         fss.longitude,
+                                         fss.height_agl,
+                                         cbsd_indoor_dep,
+                                         reliability=-1,
+                                         freq_mhz=interference.FREQ_PROP_MODEL_MHZ)
+
+  return db_loss, incidence_angle
+
+def convertAmslToAgl(latitude, longitude, height_amsl):
+  """ Convert AMSL height to AGL height."""
+
+  altitude_cbsd = wf_itm.terrainDriver.GetTerrainElevation(latitude, longitude)
+  return height_amsl - altitude_cbsd
+
+def getAntennaGainFssReceiver(fss_entity, incidence_angle):
+  """Get antenna gain of FSS receiver determined using FSS antenna reference model.
+
+  Args:
+    cbsd: A CbsdData object
+    incidence_angle: Incidence angle as identified by ITM model.
+  Returns:
+    Antenna gain of FSS in dBi.
+  """
+
+  #Invoke FSS antenna reference model.
+  fss_ant_gain = antenna.GetFssAntennaGains(
+                              incidence_angle.hor_rx, incidence_angle.ver_rx,
+                              fss_entity.pointing_azimuth,
+                              fss_entity.pointing_elevation,
+                              fss_entity.max_gain_dbi)
+  return fss_ant_gain
+
+def getHighestGrantOfCbsds(cbsds):
+  """Creates a new list of grants for oobe_calculation
```

Comment:
This doesn't really provide a list of grants.

Reply:
Modified   as per new functionality